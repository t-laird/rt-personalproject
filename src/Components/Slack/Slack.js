import React, { Component } from 'react';
import './Slack.css';
import { connect } from 'react-redux';
import validateSlackId from '../../helpers/fetches/validateSlackId/validateSlackId';

class Slack extends Component {
  constructor() {
    super();
    this.state = {
      slackId: '',
      message: null
    };
  }

  handleInput = (e) => {
    const {name, value} = e.target;
    
    this.setState({
      [name]: value
    });
  }

  handleSubmit = async () => {
    const { slackId } = this.state;
    const validateInput = await validateSlackId(slackId);

    if (validateInput.status === 'success') {
      this.setState({
        message: <h5>Your Slack id <span>{this.state.slackId}</span> was successfully added to your Snap Ninja account! Try <span>/snap help</span> for help using Snap Ninja from slack!</h5>
      });
    } else {
      this.setState({
        message: <h5>You Slack id <span>{this.state.slackId}</span> could not be added. Are you sure you put in the right information? Type /snap in slack to get your ID!</h5>
      });
    }
  }

  componentDidMount() {
    if (!this.props.User.name) {
      this.props.history.push('/login');
    }
  }

  componentContent = () => {
    if (this.props.User.slack_id) {
      return (
        <div className="Slack">
          <div className="slack-header">
            <h4>CONNECT TO SLACK</h4>
          </div>
          <h6>Your Slack ID is {this.props.User.slack_id}</h6>
          <h5>You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try <span>/snap help</span>.</h5>
        </div>
      );
    }

    return (
      <div className="Slack">
        <div className="slack-header">
          <h4>CONNECT TO SLACK</h4>
        </div>
        <h5>Type /snap anywhere in Slack to get your User Id! (case sensitive)</h5>
        <input 
          type="text" 
          value={this.state.slackId} 
          placeholder="Slack ID"
          onChange={(e) => {this.handleInput(e)}}
          name="slackId" /> <br/>
        <button onClick={this.handleSubmit}>UPDATE</button>
        {this.state.message}
      </div>
    );
  }

  render() {
    return this.componentContent();
  }
}

export const mapStateToProps = ( store ) => ({
  User: store.User
});


export default connect(mapStateToProps, null)(Slack);