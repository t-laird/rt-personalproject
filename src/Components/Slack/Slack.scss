@import '../../Variables.scss';

.Slack {

  @include center-box;
  background-color: $c6;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  .slack-header {
    @include box-header;
    background-color: $c1;

    h4 {
      color: $c7;
      font-weight: 900;
      text-align: center;
    }
  }

  h5 {
    color: $c3;
    font-weight: 900;
    margin: 30px 0px;
    line-height: 1.3;
  }

	input, button {
		overflow: visible;
    border: none;
    display: block;
    height: 60px;
    font-size: 24px;
    margin: 0px 0px 30px;
    text-align: center;
	}

	input {
		width: 250px;
		background-color: $c7;
    border-bottom: 8px solid $c5;
    color: $c3;
    margin-right: 20px;

		&::placeholder {
			color: $c1;
      font-weight: 900;
		};
	}

  button {
    height: 52px;
    width: 149px;
    color: #fff;
    background-color: $c1;
    cursor: pointer;
    transition: background-color .15s ease-in-out;
    padding-top: 2px;
    border: none;
    display: block;
    font-size: 20px;

    &:hover {
      background-color: darken($c1, 12%);
    }
  }

  h5 {
    padding: 10px 30px 20px;
    margin: 0px;
    overflow-wrap: break-word;
    max-width: 100%;
    span {
      font-size: inherit;
      font-family: inherit;
      color: $c1;
      padding: 0px 5px;
    }
  }

  h6 {
    color: $c1;
    margin: 10px 0px 0px;
  }
}
