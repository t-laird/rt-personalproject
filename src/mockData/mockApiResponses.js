export const getGroupSettingsApiResponse = {
  status: 'success',
};

export const getGroupTransactionDataApiResponse = {
	status: 'success'
};