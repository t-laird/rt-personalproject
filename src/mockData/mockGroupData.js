export const mockGroupData = {
	administrator_id: 95,
	created_date: "2017-10-18T18:34:30.017Z",
	group_id: 1,
	group_name: 'Turing',
	group_passphrase: "9139z",
	weekly_points: 100
}