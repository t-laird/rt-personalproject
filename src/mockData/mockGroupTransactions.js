const mockGroupTransactions = [
  {transactions: [
    {
      created_time: 1514939019586,
      event_id: 9,
      event_time: "2018-01-03T00:23:39.627Z",
      group_id: 1,
      note: "You totally killed it on your lightning talk this morning!  I loved it.",
      point_value: 15,
      receive_id: 22,
      received_name: "Scott Adams",
      send_id: 24,
      send_name: "Jim Davis"
    },
    {
      created_time: 1514939019586,
      event_id: 2,
      event_time: "2018-01-03T00:23:39.612Z",
      group_id: 1,
      note: "Thanks!",
      point_value: 10,
      receive_id: 59,
      received_name: "Bo Jackson",
      send_id: 43,
      send_name: "Deion Sanders"
    }
  ]}
]

export default mockGroupTransactions;