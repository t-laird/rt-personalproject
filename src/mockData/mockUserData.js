export const mockUserData = {
  authrocket_id: "usr_0vXWX9jSRkbUjLGsOb9S8r",
  created_date: "2017-11-18T18:34:30.017Z",
  email: "cpavin@usopen.com",
  group_id: 1,
  name: "Corey Pavin",
  user_id: 95
};

