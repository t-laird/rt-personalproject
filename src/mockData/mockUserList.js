export const mockUserList = [
  {
    authrocket_id: "jfnxcio42393",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "abcf@gmail.com",
    group_id: 1,
    name: "Mike Jenkins",
    user_id: 2
  },
  {
    authrocket_id: "lksdfjn23413",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "qwi123@gmail.com",
    group_id: 1,
    name: "Josh Brown",
    user_id: 3
  },
  {
    authrocket_id: "xmcv3941nv",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "293xcnv@gmail.com",
    group_id: 1,
    name: "Alan Michaelson",
    user_id: 4
  },
  {
    authrocket_id: "3490xxa1j",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "nc-1n23a@gmail.com",
    group_id: 1,
    name: "Sarah Hilman",
    user_id: 5    
  } 
];