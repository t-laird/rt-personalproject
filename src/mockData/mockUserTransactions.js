const mockUserTransactions = [
  {
    received: [
      {
        'send_id': 5,
        'receive_id': 8,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 1,
        'receive_id': 8,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ],
    sent: [
      {
        'send_id': 8,
        'receive_id': 4,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 8,
        'receive_id': 3,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ]
  }
];

export default mockUserTransactions;

