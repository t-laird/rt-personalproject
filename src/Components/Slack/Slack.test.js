import { shallow } from 'enzyme';
import React from 'react';
import { Slack, mapStateToProps } from './Slack';
import { mockUserData } from '../../mockData/mockUserData';

describe('Slack component tests', () => {
  let mockProps;
  let renderedSlack;
  beforeEach(() => {
    mockProps = {
      User: mockUserData
    };
    renderedSlack = shallow(<Slack {...mockProps}/>);
  });

  it('should match the snapshot', () => {
    expect(renderedSlack).toMatchSnapshot();
  });

  it('should redirect to the login page if there is no signed in user', () => {
    const newMockProps = {
      User: {},
      history: []
    };

    renderedSlack = shallow(<Slack {...newMockProps} />);

    const expectedHistory = '/snap-ninja/login';

    expect(renderedSlack.instance().props.history).toContain(expectedHistory);
  });

  it('should indicate thatuser has already linked slack account when appropriate', () => {
    renderedSlack.setProps({
      User: {name: 'test', slack_id: 'truthystring'}
    });

    const expectedText = 'You have already connected your Slack and Snap Ninja accounts!';

    expect(renderedSlack.find('h5').text()).toContain(expectedText);
  });

  it('should show a message explaining how to get the slack id if appropriate', () => {
    renderedSlack.setProps({
      User: {name: 'test', slack_id: null}
    });

    const expectedText = 'anywhere in Slack to get your User Id.';

    expect(renderedSlack.find('h5').text()).toContain(expectedText);
  });

  it('should attempt to validate the slack id passed in if handleSubmit is called', () => {
    global.localStorage = {
      getItem: () => JSON.stringify('sljfdsldfj')
    };
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve(
      {
        json: () => Promise.resolve({
          status: 'success'
        })
      }
    ));

    renderedSlack.instance().handleSubmit();

    expect(window.fetch).toHaveBeenCalled();
  });
});

describe('mapStateToProps tests', () => {
  it('should map the User to props', () => {
    const mockStore = {
      User: mockUserData
    };

    const result = mapStateToProps(mockStore);

    expect(result.User).toEqual(mockStore.User);
  });
});