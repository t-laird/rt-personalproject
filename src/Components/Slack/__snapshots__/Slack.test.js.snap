// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slack component tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slack
    User={
        Object {
            "authrocket_id": "usr_0vXWX9jSRkbUjLGsOb9S8r",
            "created_date": "2017-11-18T18:34:30.017Z",
            "email": "cpavin@usopen.com",
            "group_id": 1,
            "name": "Corey Pavin",
            "slack_id": "lkjsdf0234",
            "user_id": 95,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="slack-header"
>
          <h4>
                    CONNECT TO SLACK
          </h4>
</div>,
        <h6>
          Your Slack ID is 
          lkjsdf0234
</h6>,
        <h5>
          You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try 
          <span>
                    /snap help
          </span>
          .
</h5>,
      ],
      "className": "Slack",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            CONNECT TO SLACK
</h4>,
          "className": "slack-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CONNECT TO SLACK",
          },
          "ref": null,
          "rendered": "CONNECT TO SLACK",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Your Slack ID is ",
            "lkjsdf0234",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Your Slack ID is ",
          "lkjsdf0234",
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try ",
            <span>
              /snap help
</span>,
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "/snap help",
            },
            "ref": null,
            "rendered": "/snap help",
            "type": "span",
          },
          ".",
        ],
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="slack-header"
>
            <h4>
                        CONNECT TO SLACK
            </h4>
</div>,
          <h6>
            Your Slack ID is 
            lkjsdf0234
</h6>,
          <h5>
            You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try 
            <span>
                        /snap help
            </span>
            .
</h5>,
        ],
        "className": "Slack",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              CONNECT TO SLACK
</h4>,
            "className": "slack-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CONNECT TO SLACK",
            },
            "ref": null,
            "rendered": "CONNECT TO SLACK",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Your Slack ID is ",
              "lkjsdf0234",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Your Slack ID is ",
            "lkjsdf0234",
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try ",
              <span>
                /snap help
</span>,
              ".",
            ],
          },
          "ref": null,
          "rendered": Array [
            "You have already connected your Slack and Snap Ninja accounts! If you need help using Snap Ninja slash commands in Slack, try ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "/snap help",
              },
              "ref": null,
              "rendered": "/snap help",
              "type": "span",
            },
            ".",
          ],
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
